webui/react/src/hooks/useSettings.ts

```diff
@@ -280,6 +297,34 @@ const useSettings = <T>(config: SettingsConfig, options?: SettingsHookOptions):
     updateSettings(newSettings);
   }, [ config.settings, updateSettings ]);
 
+  const fetchUserSetting = useCallback(async () => {
+
+    if(!user) return;
+    const userSettingResponse = await getUserSetting({ userId: user.id });
+    userSettingResponse.settings.forEach(setting => {
```

Comment:
it might be nice to be able to query for specific settings at some point, to avoid processing the whole thing.

Reply:
I agree, but I wrote like this because `basePath` is not exposed to this level, so even we can query with a path, we do not know the path outside of `Storage` component.